packages/aws-cdk-lib/aws-cloudwatch/lib/alarm.ts

```diff
@@ -208,26 +248,71 @@ export class Alarm extends AlarmBase {
       ...metricProps,
     });
 
+    this.onCfnAlarmCreated(alarm);
+
     this.alarmArn = this.getResourceArnAttribute(alarm.attrArn, {
       service: 'cloudwatch',
       resource: 'alarm',
       resourceName: this.physicalName,
       arnFormat: ArnFormat.COLON_RESOURCE_NAME,
     });
+
     this.alarmName = this.getResourceNameAttribute(alarm.ref);
 
     this.metric = props.metric;
-    const datapoints = props.datapointsToAlarm || props.evaluationPeriods;
-    this.annotation = {
-      // eslint-disable-next-line max-len
-      label: `${this.metric} ${OPERATOR_SYMBOLS[comparisonOperator]} ${props.threshold} for ${datapoints} datapoints within ${describePeriod(props.evaluationPeriods * metricPeriod(props.metric).toSeconds())}`,
-      value: props.threshold,
-    };
+    this.annotation = this.createHorizontalAnnotation(comparisonOperator, props);
 
     for (const [i, message] of Object.entries(this.metric.warningsV2 ?? {})) {
       Annotations.of(this).addWarningV2(i, message);
     }
   }
+  protected validateComparisonOperator(comparisonOperator: ComparisonOperator): void {
+    if (Alarm.isAnomalyDetectionOperator(comparisonOperator)) {
+      throw new ValidationError(
+        'Anomaly detection operators require to use the construct AnomalyDetectionAlarm.',
+        this,
+      );
+    }
+  }
+  protected validateThreshold(threshold?: number): void {
+    // For standard alarms, we need a threshold
+    if (threshold === undefined) {
+      throw new ValidationError('threshold must be specified for standard alarms', this);
+    }
+  }
+
+  protected onCfnAlarmCreated(_alarm: CfnAlarm): void {
+    // Do nothing in Alarm construct.
+  }
+
+  /**
+   * Gets the threshold value for CloudFormation alarm properties.
+   *
+   * @param props - The alarm properties
+   * @returns The threshold value to use, or undefined if none is applicable
+   */
+  protected getThresholdForCfnAlarmProps(props: AlarmProps): number | undefined {
+    return props.threshold;
+  }
+
+  /**
+   * Gets the thresholdMetricId for CloudFormation alarm properties.
+   *
+   * @param _metrics - The array of metric data queries to validate against
+   * @returns The threshold metric ID to use, or undefined if none is applicable
+   */
+  protected getThresholdMetricIdForCfnAlarmProps(_metrics: CfnAlarm.MetricDataQueryProperty[]): string | undefined {
+    return undefined;
+  }
```

Comment:
I don't like any of this. Can we not pass in the properties into the `super()` constructor?